### PR TITLE
feat(oidc): add spotify provider

### PR DESCRIPTION
**Pull request by aeneasr**
## Related issue(s)

Closes #1985

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

File: selfservice/strategy/oidc/provider_spotify.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+func (g *ProviderSpotify) AuthCodeURLOptions(r ider) []oauth2.AuthCodeOption {
+	return []oauth2.AuthCodeOption{}
+}
+
+func (g *ProviderSpotify) Claims(ctx context.Context, exchange *oauth2.Token) (*Claims, error) {
+	grantedScopes := stringsx.Splitx(fmt.Sprintf("%s", exchange.Extra("scope")), " ")
+	for _, check := range g.Config().Scope {
+		if !stringslice.Has(grantedScopes, check) {
+			fmt.Println("SCOPE MISSING", check, fmt.Sprintf("%#v", exchange.Extra("scope")))
```
Comment (aeneasr) [inline selfservice/strategy/oidc/provider_spotify.go:63]:
```suggestion
```

File: selfservice/strategy/oidc/provider_spotify.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+	claims := &Claims{
+		Subject:   user.ID,
+		Issuer:    spotify.Endpoint.TokenURL,
+		Name:      user.DisplayName,
+		Nickname:  user.DisplayName,
+		Email:     user.Email,
+		Picture:   userPicture,
+		Profile:   user.ExternalURLs["spotify"],
+		Birthdate: user.Birthdate,
```
Comment (bradleyess) [inline selfservice/strategy/oidc/provider_spotify.go:91]: Hey there! I'm looking to make a contribution here in order to get Country from the Spotify API. Would it be simple as adding `Country: user.country` into the Claims struct here?

My apologies - I'm not familiar with the providers or Go. But I'd love to make my first very simple contribution to Kratos.

Should this change be accepted, what is the average turn-around on a release for a non-breaking change like this making it to Ory Cloud? This is a requirement for a system I am building.

**Review by bradleyess — COMMENTED** [review]
👋🏼 sorry for reviving such an old PR here.

File: selfservice/strategy/oidc/provider_spotify.go (hunk 2 above)
Comment (zepatrik) [reply selfservice/strategy/oidc/provider_spotify.go:91]: > Would it be simple as adding Country: user.country into the Claims struct here?

Not sure, but I suggest you try it with a local Kratos instance and your patch. Make sure to check the spotify docs on the exact key and scopes needed.

> what is the average turn-around on a release for a non-breaking change like this making it to Ory Cloud?

Days, but we can prioritize this when requested.

Comment (zepatrik) [reply selfservice/strategy/oidc/provider_spotify.go:91]: > The country of the user, as set in the user's account profile. An [ISO 3166-1 alpha-2 country code](http://en.wikipedia.org/wiki/ISO_3166-1_alpha-2). This field is only available when the current user has granted access to the [user-read-private](https://developer.spotify.com/documentation/general/guides/authorization-guide/#list-of-scopes) scope.

from https://developer.spotify.com/documentation/web-api/reference/#/operations/get-current-users-profile

**Pull request merged**